accord-core/src/main/java/accord/utils/RandomSource.java

```diff
@@ -125,31 +222,53 @@ else if (delta > 0L)
         }
         return result;
     }
-
-    default LongStream longs()
+    default long nextBiasedLong(long minInclusive, long mean, long maxExclusive)
     {
-        return LongStream.generate(this::nextLong);
+        // this is diff behavior than ThreadLocalRandom, which returns nextInt
+        if (minInclusive >= maxExclusive)
+            throw new IllegalArgumentException(String.format("Min (%s) should be less than max (%d).", minInclusive, maxExclusive));
+
+        long range = Math.max(maxExclusive - mean, mean - minInclusive) * 2;
```

Comment:
there are no checks that `mean` is within the range, I tried the following and found that we don't respect min/max

```
nextBiasedLong(1ms, 500000ms, 1s)
```

```
Expecting:
	 <5783312L> // 5.78s
	to be between:
	 [1000L, 999999L]
```

Not sure if you intend for this to be valid, I am finding a hard time understanding the data in this case, so if you do mean for this to be supported can you document the semantics?

Reply:
I have updated to exclude means that are out of range.